x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/NegMapper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.planner;
+
+import org.elasticsearch.compute.operator.EvalOperator;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.Neg;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+abstract class NegMapper extends EvalMapper.ExpressionMapper<Neg> {
+    static final EvalMapper.ExpressionMapper<?> NEG_MAPPER = new NegMapper(
+        org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.NegIntsEvaluator::new,
+        org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.NegLongsEvaluator::new,
+        org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.NegDoublesEvaluator::new
+    ) {
+    };
+
+    private final BiFunction<Source, EvalOperator.ExpressionEvaluator, EvalOperator.ExpressionEvaluator> ints;
+
+    private final BiFunction<Source, EvalOperator.ExpressionEvaluator, EvalOperator.ExpressionEvaluator> longs;
+    private final Function<EvalOperator.ExpressionEvaluator, EvalOperator.ExpressionEvaluator> doubles;
+
+    private NegMapper(
+        BiFunction<Source, EvalOperator.ExpressionEvaluator, EvalOperator.ExpressionEvaluator> ints,
+        BiFunction<Source, EvalOperator.ExpressionEvaluator, EvalOperator.ExpressionEvaluator> longs,
+        Function<EvalOperator.ExpressionEvaluator, EvalOperator.ExpressionEvaluator> doubles
+    ) {
+        this.ints = ints;
+        this.longs = longs;
+        this.doubles = doubles;
+    }
+
+    @Override
+    protected final Supplier<EvalOperator.ExpressionEvaluator> map(Neg neg, Layout layout) {
+        DataType type = neg.dataType();
+        if (type.isNumeric()) {
+            var childEvaluator = EvalMapper.toEvaluator(neg.field(), layout).get();
+
+            // Unsigned longs are unsupported by choice; negating them does not make much sense without converting to long;
```

Comment:
Nit: I'd move this comment on top of LONG handling, which is "closer" to the UL type; and maybe also shorten it (IMO the reason ultimately being that conversion would require some implicit type change, less to do with subtraction consistency).
But not worth a new CI round just for this.

Reply:
Will happily move! Not sure which location you have in mind, though - a cursory search didn't turn up any locations with e.g. comments on long handling for me. Can you point me to where you would rather place this?